Mathlib/RingTheory/TensorProduct/MonoidAlgebra.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2025 Michał Mrugała. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michał Mrugała
+-/
+import Mathlib.Algebra.MonoidAlgebra.Basic
+import Mathlib.RingTheory.IsTensorProduct
+
+/-!
+# Monoid algebras are invariant under base change
+
+In this file we show that monoid algebras are stable under pushout.
+
+## TODO
+
+Additivise
+-/
+
+noncomputable section
+
+open TensorProduct
+
+namespace MonoidAlgebra
+variable {R M S A B : Type*} [CommSemiring R] [CommSemiring S] [Algebra R S] [CommMonoid M]
+variable [CommSemiring A] [Algebra R A] [CommSemiring B] [Algebra R B]
+
+variable (R A B) in
+/-- Tensoring `MonoidAlgebra R M` on the left by an `R`-algebra `A` is algebraically
+equivalent to `MonoidAlgebra A M`. -/
```

Comment:
```suggestion
/-- The base change of `MonoidAlgebra R M` to an `R`-algebra `A` is isomorphic to
`MonoidAlgebra A M` as an `A`-algebra. -/
```
I don't think "algebraically equivalent" is a term (unless you are talking about algebraic equivalence of divisors).

Edit: Also, this probably should be the docstring of `scalarTensorEquiv` and `tensorEquiv` should get its own docstring.